flocker/common/_interface.py

```diff
@@ -78,3 +80,50 @@ def invariant(value):
     invariant.__name__ = "provides_{}_invariant".format(interface_name)
 
     return invariant
+
+
+@attributes([
+    Attribute("unexpected_arguments", default_value=frozenset()),
+    Attribute("missing_arguments", default_value=frozenset()),
+    Attribute("missing_optional_arguments", default_value=frozenset())
+])
+class InvalidSignature(Exception):
+    """
+    See :func:`validate_signature_against_kwargs` raises clause.
+    """
+    def __str__(self):
+        return repr(self)
+
+
+def validate_signature_against_kwargs(function, keyword_arguments):
+    """
+    Validates that ``function`` can be called with keyword arguments with
+    exactly the passed in ``keyword_arguments_keys``. In this case validation
+    is verifying that the function's signature allows it to be called with
+    exactly the given keyword arguments.
+
+    :param function: The function of which to verify the signature.
+    :param set keyword_arguments_keys: A set of keyword argument names to
+        validate against the function signature.
+
+    :returns: ``None`` if the validation succeeds.
+
+    :raises InvalidSignature: If validation fails.
```

Comment:
I'll think on this. To me this seems more pythonic. Although maybe if I return a new object "signature differences" or something... I'll get back to this before I'm done.